### PR TITLE
use sets.Set to replace the array

### PR DESCRIPTION
**Pull request by ChenglongGao327**
1. use sets.Set
2. fix a typo

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/security/trustdomain/bundle.go [hunk 1]
```diff
@@ -103,11 +104,11 @@ func (t Bundle) replaceTrustDomains(principal, trustDomainFromPrincipal string)
...
 		// has a * prefix. For example, "*-td" can match with "old-td" and "new-td", but we only want
 		// to keep the principal as-is in the generated config, .i.e. *-td.
-		if !isKeyInList(newPrincipal, principalsForAliases) {
-			principalsForAliases = append(principalsForAliases, newPrincipal)
+		if !principalsForAliases.Contains(newPrincipal) {
+			principalsForAliases.Insert(newPrincipal)
 		}
 	}
-	return principalsForAliases
+	return principalsForAliases.SortedList()
```
Comment (ChenglongGao327) [inline pilot/pkg/security/trustdomain/bundle.go:111]: sort newPrincipal array, need adjust the order of test data

**Review by howardjohn — CHANGES_REQUESTED** [review]
I am suspicious that this is faster.. we need to alloc a map (slow) and sort it (slow).

if we are going to micro-optimize we should have benchmarks proving its actually an improvement IMO

File: pilot/pkg/security/trustdomain/bundle.go [hunk 2]
```diff
@@ -84,9 +85,9 @@ func (t Bundle) ReplaceTrustDomainAliases(principals []string) []string {
 // replaceTrustDomains replace the given principal's trust domain with the trust domains from the
 // trustDomains list and return the new principals.
 func (t Bundle) replaceTrustDomains(principal, trustDomainFromPrincipal string) []string {
-	principalsForAliases := []string{}
+	principalsForAliases := sets.Set{}
```
Comment (yangminzhu) [inline pilot/pkg/security/trustdomain/bundle.go:88]: In most cases there will be only a few items in the `principalsForAliases` and `t.TrustDomains` list, this likely won't be any faster especially you have to do an extra `SortedList()` in the end.

**Comment by ChenglongGao327** [comment]
yes，if there only a few items this won't be faster

**Pull request closed without merging**